.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,64 @@
+name: Build and Test
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types:
+      - opened
+      - synchronize
+jobs:
+  build-and-test:
+    strategy:
+      matrix:
+        ghc: ["8.0.2", "8.2.2", "8.4.4", "8.6.5", "8.8.3", "8.10.1"]
+        os: [ubuntu-latest]
+        include:
+          - os: macos-latest
+            ghc: "8.10.1"
+          - os: ubuntu-16.04
+            ghc: "7.10.3"
```

Comment:
@ibarrae maybe we should consider dropping support for GHC 7.10.3 is quite too old

Reply:
I'm up for it but I think that would require a bigger change in the code. What I'm trying to say is that removing this from the build matrix is fairly easy but maybe we should also consider CPP statements that we in order to support older versions?

I think that might be worth it's own PR IMHO, just to keep them small :smile:

Reply:
@ibarrae fair enough, would you mind adding a separate ticket to get rid of GHC 7.10.3 at the CI and code level, please?

Reply:
Sure, I just created a new [issue](https://github.com/stackbuilders/hapistrano/issues/160) for it :+1: